frontend/dockerfile/docs/rules/from-platform-flag-const-disallowed.md

```diff
@@ -0,0 +1,46 @@
+---
+title: FromPlatformFlagConstDisallowed
+description: FROM --platform should not use a constant value
+aliases:
+  - /go/dockerfile/rule/from-platform-flag-const-disallowed/
+---
+
+## Output
+
+```text
+FROM --platform=linux/amd64 should not use a constant value
+```
+
+## Description
+
+Specifying `--platform` in the Dockerfile `FROM` instruction forces the image to build on only one target platform. This prevents building a multi-platform image from this Dockerfile and you must build on the same platform as specified in `--platform`.
+
+The recommended approach is to:
+
+* Omit `FROM --platform` in the Dockerfile and use the `--platform` argument on the command line.
+* Use `$BUILDPLATFORM` or some other combination of variables for the `--platform` argument.
+* Stage name should include the platform, OS, or architecture name.
```

Comment:
"Stage name should include the platform, OS, or architecture name to indicate that it only contains platform-specific instructions"